easycv/models/detection/detectors/yolox/yolo_pafpn.py

```diff
@@ -1,90 +1,281 @@
 # Copyright (c) 2014-2021 Megvii Inc And Alibaba PAI-Teams. All rights reserved.
+import logging
+import math
 
 import torch
 import torch.nn as nn
 
 from easycv.models.backbones.darknet import CSPDarknet
-from easycv.models.backbones.network_blocks import BaseConv, CSPLayer, DWConv
+from easycv.models.backbones.network_blocks import (BaseConv, CSPLayer, DWConv,
+                                                    GSConv, VoVGSCSP)
+from easycv.models.backbones.repvgg_yolox_backbone import RepVGGYOLOX
+
+
+def make_divisible(x, divisor):
+    # Upward revision the value x to make it evenly divisible by the divisor.
+    return math.ceil(x / divisor) * divisor
 
 
 class YOLOPAFPN(nn.Module):
     """
     YOLOv3 model. Darknet 53 is the default backbone of this model.
     """
 
-    def __init__(
-        self,
-        depth=1.0,
-        width=1.0,
-        in_features=('dark3', 'dark4', 'dark5'),
-        in_channels=[256, 512, 1024],
-        depthwise=False,
-        act='silu',
-    ):
+    def __init__(self,
+                 depth=1.0,
+                 width=1.0,
+                 in_features=('dark3', 'dark4', 'dark5'),
+                 in_channels=[256, 512, 1024],
+                 depthwise=False,
+                 act='silu',
+                 asff_channel=16,
+                 use_att=None,
+                 expand_kernel=3,
+                 down_rate=32,
+                 use_dconv=False,
+                 use_expand=True,
+                 backbone='CSPDarknet',
+                 neck='yolo',
+                 neck_mode='all'):
         super().__init__()
-        self.backbone = CSPDarknet(depth, width, depthwise=depthwise, act=act)
+        # build backbone
+        if backbone == 'CSPDarknet':
+            self.backbone = CSPDarknet(
+                depth, width, depthwise=depthwise, act=act)
+        elif backbone == 'RepVGGYOLOX':
+            self.backbone = RepVGGYOLOX(
+                in_channels=3, depth=depth, width=width)
+        else:
+            logging.warning(
+                'YOLOX-PAI backbone must in [CSPDarknet, RepVGGYOLOX], otherwise we use RepVGGYOLOX as default'
+            )
+            self.backbone = RepVGGYOLOX(
+                in_channels=3, depth=depth, width=width)
+        self.backbone_name = backbone
+
+        # build neck
         self.in_features = in_features
         self.in_channels = in_channels
         Conv = DWConv if depthwise else BaseConv
+        self.neck = neck
```

Comment:
neck move to detection/necks/

Reply:
neck is alongwith backbone in yolox currently.